content/blog/2017/2017-02-10-declarative-html-publisher.adoc

```diff
@@ -0,0 +1,322 @@
+link:https://github.com/reiseburo/hermann[hermann project].
+I've set up a Multibranch Pipeline and point it at my repository
+the same as did it previous blog.
+Also the same as before, I've also set this Pipeline's Git configuration to
+automatically "Clean after checkout".
+
+This time we already had a Pipeline checked in.  I'll run it a few times to get
+get a baseline.
+
+image::/images/post-images/2017-02-09/scripted-build.png[Stage view, role="center"]
+image::/images/post-images/2017-02-09/scripted-rcov.png[RCov Report, role="center"]
+
+== Converting to Declarative
+
+Let's start by converting the Scripted Pipeline straight to Declarative.
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+  agent any // <1> // <2>
+  options {
+    // Keep the 10 most recent builds
+    buildDiscarder(logRotator(numToKeepStr:'10')) // <3>
+  }
+  stages {
+    stage ('Build') { // <4>
+      steps {
+        // install required gems
+        sh 'bundle install'
+
+        // build and run tests with coverage
+        sh 'bundle exec rake build spec'
+
+        // Archive the built artifacts
+        archive includes: 'pkg/*.gem'
+
+        // publish html
+        publishHTML target: [
+            allowMissing: false,
+            alwaysLinkToLastBuild: false,
+            keepAll: true,
+            reportDir: 'coverage',
+            reportFiles: 'index.html',
+            reportName: "RCov Report"
+          ]
+      }
+    }
+  }
+}
+// Scripted //
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]]) // <4>
+
+node { // <1>
+  stage ('Build') { // <4>
+
+    // Checkout
+    checkout scm // <2>
+
+    // install required gems
+    sh 'bundle install'
+
+    // build and run tests with coverage
+    sh 'bundle exec rake build spec'
+
+    // Archive the built artifacts
+    archive includes: 'pkg/*.gem'
+
+    // publish html
+    publishHTML [
+        allowMissing: false,
+        alwaysLinkToLastBuild: false,
+        keepAll: true,
+        reportDir: 'coverage',
+        reportFiles: 'index.html',
+        reportName: "RCov Report"
+      ]
+
+  }
+}
+----
+<1> Select where to run this Pipeline, in this case "any" agent, regardless of label.
+<2> Declarative automatically performs a checkout of source code on the agent,
+    whereas Scripted Pipeline users must explicitly call `checkout scm`.
+<3> Set the Pipeline option to preserve the ten most recent runs.
+This overrides the default behavior from the Multibranch parent of this Pipeline.
+<4> Run the "Build" stage.
+
+image::/images/post-images/2017-02-09/declarative-build.png[Stage view, role="center"]
+
+Now that we have this Pipeline in Declarative form, let's take a minute to do a
+little clean up.  We'll split out the `bundle` actions a little more and move
+steps into logically grouped stages.  Rather than having one monolithic "Build"
+stage, we'll have details for each stage.  As long as we're prettying things
```

Comment:
Nah. This is okay as is. 
"As long as we're" is closer in meaning to "Since we're already".